src/compat/_internal/isIndex.ts

```diff
@@ -1,9 +1,11 @@
 const IS_UNSIGNED_INTEGER = /^(?:0|[1-9]\d*)$/;
 
-export function isIndex(value: PropertyKey) {
+export function isIndex(value: PropertyKey, length?: number): boolean {
```

Comment:
I understand that this function is already defined in `lodash`, but I think receiving the `length` as an argument is very confusing. For example, I can't predict what will happen for the following code:

```tsx
isIndex(value, 3);
```

So I wish we move the clamping logic outside of `isIndex`. What do you think?

Reply:
That seems like a good idea, so I applied it to the code as well!